tid2md.py

```diff
@@ -193,7 +193,7 @@ def write_markdown_file(lines: list, md_file: Path) -> bool:
                 # External links
                 # [[text|url]] → [text](url)
                 # [[url]] → <url>
-                line = re_link.sub(parse_link, line)
+                line = re_link.sub(lambda m: parse_link(m, use_angles), line)
```

Comment:
I guess you turned this into a lambda (again) by accident while rebasing!? Or does it bring any advantage over `re_link.sub(parse_link(m, use_angles), line)`

Reply:
This was intentional. You're throwing me for a loop here (again, been some years since I used Python), so I had to double-check. Running `re_link.sub(parse_link(m, use_angles), line)` in the python REPL produces a NameError as I expected, because "m" is not a defined variable; running the script itself throws an "'NoneType' object has no attribute 'group'" error, because "m" is never set to anything. The delegate function that re.sub is waiting for must take one argument only, it seems.

The only thing I can come up with to avoid wrapping parse_link in a lambda is to make it curryable, e.g.
```
def parse_link(use_angles: bool) -> str:
    def parse_inner(match: re.Match):
        # Processing unchanged, returns a string
    return parse_inner
```
This is kind of exactly the same as wrapping it in a lambda though, but it allows us to use `line = re_link.sub(parse_link(use_angles), line)`. I'm quite certain this isn't what you had in mind though... or was it?

In short, I'm not exactly sure what you mean, your proposed alternative doesn't seem to work. Could you elaborate? (Don't tell me it's my windows again, I'm running anaconda like a sane person. :P)

Reply:
OK, now I get it. I was even wondering why the first argument of `sub` is a function at all (instead of a string). Now I see that this first argument is like the second argument in `re.sub` which can be either a string or a function. I didn't even notice that your were passing the `parse_link` function before (and not a function call).